hyperspy/io_plugins/_hierarchical.py

```diff
@@ -626,7 +634,17 @@ def overwrite_dataset(cls, group, data, key, signal_axes=None,
                     del group[key]
 
         _logger.info(f"Chunks used for saving: {chunks}")
-        cls._store_data(data, dset, group, key, chunks)
+        if data.dtype == np.dtype('O'):
```

Comment:
To enforce declaring signal as being ragged, I think we should use this approach only when the signal is ragged:
```suggestion
        if self.signal.ragged:
```

Reply:
I don't know if that is the best behavior right now.  Currently the ragged attribute isn't firmly typed i.e. you can have a signal which has an object dtype but isn't ragged.  It probably shouldn't be the case going forward but for now I think that would cause some edge case signal to not save.

Reply:
Yes, this is the motivation to enforce the use of `ragged` signal when necessary and special case `ragged` processing only when the signal is explicitly set as ragged, otherwise this is speculative.

If I understand/remember correctly, the issue is with saving ragged array, not `object` type array? If so, I think that using `self.signal.ragged` is more appropriate.